### PR TITLE
Tweak for Ram usage

### PR DESCRIPTION
**Pull request by etrinh**
Add KNX_NO_DEFAULT_UART to avoid default uart assignation for ArduinoPlatform and derivatives, with associated defines (HWSERIAL_NONE for stm32) to recover RAM from unused RX/TX buffers

Remove ArduinoPlatform(HardwareSerial* knxSerial) constructor, seems not useable by KnxFacade

### CODE REVIEW

File: src/arduino_platform.cpp [hunk 1]
```diff
@@ -6,9 +6,11 @@
 #include <SPI.h>
 #endif
 
+#ifndef KNX_NO_PRINT
 Stream* ArduinoPlatform::SerialDebug = &Serial;
+#endif
 
-ArduinoPlatform::ArduinoPlatform(HardwareSerial* knxSerial) : _knxSerial(knxSerial)
```
Comment (thelsing) [inline src/arduino_platform.cpp:11]: This constructor should stay. Why not create an other with no arguments.

**Comment by etrinh** [comment]
Is it what you expected?

**Comment by thelsing** [comment]
Nearly. Just keep the constructors with HardwareSerial in the derived classes of ArduinoPlatform, so people can use them if they want to.

**Comment by thelsing** [comment]
I totally forgot about this. Sorry. Feel free to give me a ping next time.

**Pull request merged**